homeassistant/components/homekit/type_humidifiers.py

```diff
@@ -0,0 +1,240 @@
+        self._hk_device_class = HC_HASS_TO_HOMEKIT_DEVICE_CLASS[device_class]
+
+        self._target_humidity_char_name = HC_DEVICE_CLASS_TO_TARGET_CHAR[
+            self._hk_device_class
+        ]
+        self.chars.append(self._target_humidity_char_name)
+
+        serv_humidifier_dehumidifier = self.add_preload_service(
+            SERV_HUMIDIFIER_DEHUMIDIFIER, self.chars
+        )
+
+        # Current and target mode characteristics
+        self.char_current_humidifier_dehumidifier = serv_humidifier_dehumidifier.configure_char(
+            CHAR_CURRENT_HUMIDIFIER_DEHUMIDIFIER, value=0
+        )
+        self.char_target_humidifier_dehumidifier = serv_humidifier_dehumidifier.configure_char(
+            CHAR_TARGET_HUMIDIFIER_DEHUMIDIFIER, value=0,
+        )
+
+        # Current and target humidity characteristics
+        self.char_current_humidity = serv_humidifier_dehumidifier.configure_char(
+            CHAR_CURRENT_HUMIDITY, value=0
+        )
+
+        max_humidity = state.attributes.get(ATTR_MAX_HUMIDITY, DEFAULT_MAX_HUMIDITY)
+        max_humidity = round(max_humidity)
+        max_humidity = min(max_humidity, 100)
+
+        min_humidity = state.attributes.get(ATTR_MIN_HUMIDITY, DEFAULT_MIN_HUMIDITY)
+        min_humidity = round(min_humidity)
+        min_humidity = max(min_humidity, 0)
+
+        self.char_target_humidity = serv_humidifier_dehumidifier.configure_char(
+            self._target_humidity_char_name,
+            value=45,
+            properties={
+                PROP_MIN_VALUE: min_humidity,
+                PROP_MAX_VALUE: max_humidity,
+                PROP_MIN_STEP: 1,
+            },
+        )
+
+        # Active/inactive characteristics
+        self.char_active = serv_humidifier_dehumidifier.configure_char(
+            CHAR_ACTIVE, value=False
+        )
+
+        self.async_update_state(state)
+
+        serv_humidifier_dehumidifier.setter_callback = self._set_chars
+
+        self.linked_humidity_sensor = self.config.get(CONF_LINKED_HUMIDITY_SENSOR)
+        if self.linked_humidity_sensor:
+            humidity_state = self.hass.states.get(self.linked_humidity_sensor)
+            if humidity_state:
+                self._async_update_current_humidity(humidity_state)
+
+    async def run_handler(self):
+        """Handle accessory driver started event.
+
+        Run inside the Home Assistant event loop.
+        """
+        if self.linked_humidity_sensor:
+            async_track_state_change_event(
+                self.hass,
+                [self.linked_humidity_sensor],
+                self.async_update_current_humidity_event,
+            )
+
+        await super().run_handler()
+
+    @callback
+    def async_update_current_humidity_event(self, event):
+        """Handle state change event listener callback."""
+        self._async_update_current_humidity(event.data.get("new_state"))
+
+    def _async_update_current_humidity(self, new_state):
```

Comment:
```suggestion
    @callback
    def _async_update_current_humidity(self, new_state):
```

Reply:
Why is it needed, by the way? We don't register this function as a callback, only `async_update_current_humidity_event`

Reply:
I missed you were only calling it from `async_update_current_humidity_event`.  Either way its safe to call from async.